src/catalog/database_catalog.cpp

```diff
@@ -1324,6 +1325,102 @@ std::vector<std::pair<common::ManagedPointer<storage::index::Index>, const Index
   return index_objects;
 }
 
+sequence_oid_t DatabaseCatalog::CreateSequence(const common::ManagedPointer<transaction::TransactionContext> txn,
+                                               const std::string &name) {
+  if (!TryLock(txn)) return INVALID_SEQUENCE_OID;
+  const sequence_oid_t sequence_oid = static_cast<sequence_oid_t>(next_oid_++);
+  return CreateSequence(txn, sequence_oid, name) ? sequence_oid : INVALID_SEQUENCE_OID;
+}
+
+bool DatabaseCatalog::CreateSequence(common::ManagedPointer<transaction::TransactionContext> txn,
+                                          sequence_oid_t sequence_oid, const std::string &name) {
+  // First, insert into pg_class
+  auto *const class_insert_redo = txn->StageWrite(db_oid_, postgres::CLASS_TABLE_OID, pg_class_all_cols_pri_);
+  auto *const class_insert_pr = class_insert_redo->Delta();
+
+  // Write the sequence_oid into the PR
+  auto sequence_oid_offset = pg_class_all_cols_prm_[postgres::RELOID_COL_OID];
+  auto *sequence_oid_ptr = class_insert_pr->AccessForceNotNull(sequence_oid_offset);
+  *(reinterpret_cast<sequence_oid_t *>(sequence_oid_ptr)) = sequence_oid;
+
+  const auto name_varlen = storage::StorageUtil::CreateVarlen(name);
+
+  // Write the name into the PR
+  const auto name_offset = pg_class_all_cols_prm_[postgres::RELNAME_COL_OID];
+  auto *const name_ptr = class_insert_pr->AccessForceNotNull(name_offset);
+  *(reinterpret_cast<storage::VarlenEntry *>(name_ptr)) = name_varlen;
+
+  // Write the ns_oid into the PR
+  const auto ns_offset = pg_class_all_cols_prm_[postgres::RELNAMESPACE_COL_OID];
+  // TODO(zianke): Write the ns_oid into the PR
```

Comment:
Would be helpful to know what needs to happen (if anything) before you can write ns_old into the PR.